Tensile/Source/lib/source/ContractionSolution.cpp

```diff
@@ -1146,6 +1207,22 @@ namespace Tensile
         if(Debug::Instance().printWinningKernelName())
             std::cout << "Running kernel: " << this->KernelName() << std::endl;
 
+        if(Debug::Instance().printKernelCommonParams())
+        {
+            std::cout << "Kernel (" << this->KernelName() << ") Params:" << std::endl;
+
+            vector4<std::uint32_t> matrixInst;
+            if (this->getMatrixInstructionFromKernelName(matrixInst))
+                std:: cout <<  std::right << std::setw(30) << "MatrixInstruction: " << matrixInst << std::endl;
+
+            std::string GSUAlg;
+            if (this->getGSUAlgorithmFromKernelName(GSUAlg))
+                std:: cout <<  std::right << std::setw(30) << "GSUAlgorithm: " << GSUAlg << std::endl;
+
+            if (this->getMatrixInstructionFromKernelName(matrixInst))
```

Comment:
Would you avoid calling same function twice?
Please store the result of the first call.